cli/internal/run/dry_run.go

```diff
@@ -68,15 +69,22 @@ func DryRun(
 		tracker,
 		rs,
 		base,
-		turboCache,
 	)
 
 	if err != nil {
 		return err
 	}
 
+	// We walk the graph with no concurrency.
+	// Populating the cache state is parallelizable.
+	// Do this _after_ walking the graph.
+	withCacheState, err := checkCacheState(turboCache, taskSummaries)
```

Comment:
Looks like `taskSummaries` is being mutated, so the return/reassign isn't necessary. Can we just comment that `taskSummaries` is being mutated instead of having faux immutability? What's idiomatic here btw? E.g. `sort.Strings()` doesn't try to hide that it's mutating the input.